### PR TITLE
Twilio R1 integration

### PR DESCRIPTION
**Pull request by harryhcs**
- Get data from D1 to send to whatsapp 
- Save and send vCards to and from R2 bucket

### CODE REVIEW

File: experiments/rsc/scripts/runDevServer.mts [hunk 1]
```diff
@@ -14,6 +14,7 @@ import {
 import { buildVendorBundles } from "./buildVendorBundles.mjs";
 import { codegenTypes } from "./codegenTypes.mjs";
 import { dispatchNodeRequestToMiniflare } from "./lib/requestUtils.mjs";
+import { getR2Buckets } from "./lib/getR2Nuckets.js";
```
Comment (justinvdm) [inline experiments/rsc/scripts/runDevServer.mts:17]: Do we use this?

File: experiments/rsc/src/worker.tsx [hunk 2]
```diff
@@ -36,8 +36,35 @@ export default {
 
       setupDb(env);
 
-      // incoming twilio request
+      // The worker access the bucket and returns it to the user, we dont let them access the bucket directly
+      if (request.method === "GET" && url.pathname.startsWith("/bucket/")) {
+        const filename = url.pathname.slice("/bucket/".length);
+        const object = await env.valley_directory_r2.get(filename);
```
Comment (justinvdm) [inline experiments/rsc/src/worker.tsx:42]: Nice!

**Review by justinvdm — APPROVED** [review]
Very cool!

File: experiments/rsc/scripts/lib/getR2Nuckets.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { readWranglerConfig } from "./readWranglerConfig";
+
+export const getR2Buckets = async () => {
+  const config = await readWranglerConfig();
+  return config.r2_buckets.map((bucket: { binding: string }) => bucket.binding);
+};
```
[the remark this answers is not in the document]
Comment (harryhcs) [reply experiments/rsc/scripts/lib/getR2Nuckets.ts:6]: oops!

File: experiments/rsc/scripts/runDevServer.mts (hunk 1 above)
Comment (harryhcs) [reply experiments/rsc/scripts/runDevServer.mts]: removed this, thanks

**Pull request merged**